Re-enable redox build

See #4851 

https://github.com/RustPython/RustPython/actions/runs/4616329249/jobs/8161219149

```
    Checking nix v0.26.2 (https://github.com/coolreader18/nix?branch=0.26.2-redox#f0c05e18)
error[E0425]: cannot find value `IPPROTO_RAW` in crate `libc`
   --> /github/home/.cargo/git/checkouts/nix-d21d02d10a5076ec/f0c05e1/src/sys/socket/mod.rs:146:17
    |
146 |     Raw = libc::IPPROTO_RAW,
    |                 ^^^^^^^^^^^ help: a constant with a similar name exists: `IPPROTO_IP`
    |
   ::: /github/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.140/src/unix/mod.rs:304:1
    |
304 | pub const IPPROTO_IP: ::c_int = 0;
    | ----------------------------- similarly named constant `IPPROTO_IP` defined here

For more information about this error, try `rustc --explain E0425`.
error: could not compile `nix` due to previous error
warning: build failed, waiting for other jobs to finish...
redoxer cargo: exit status: 101
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -230,12 +230,12 @@ jobs:
       - name: Check compilation for freeBSD
         run: cargo check --target x86_64-unknown-freebsd
 
-      # - name: Prepare repository for redox compilation
-      #   run: bash scripts/redox/uncomment-cargo.sh
-      # - name: Check compilation for Redox
-      #   uses: coolreader18/redoxer-action@v1
-      #   with:
-      #     command: check
+      - name: Prepare repository for redox compilation
+        run: bash scripts/redox/uncomment-cargo.sh
+      - name: Check compilation for Redox
+        uses: coolreader18/redoxer-action@v1
+        with:
+          command: check
 
   snippets_cpython:
     if: ${{ !contains(github.event.pull_request.labels.*.name, 'skip:ci') }}

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -130,5 +130,4 @@ lto = "thin"
 # REDOX START, Uncomment when you want to compile/check with redoxer
 # nix = { git = "https://github.com/coolreader18/nix", branch = "0.26.2-redox" }
 # errno = { git = "https://github.com/coolreader18/rust-errno", branch = "0.2.8-redox" }
-# libc = { git = "https://github.com/rust-lang/libc" }
 # REDOX END

